nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-runtime/src/main/java/org/apache/nifi/NiFi.java

```diff
@@ -55,6 +59,25 @@
 
     public static final String BOOTSTRAP_PORT_PROPERTY = "nifi.bootstrap.listen.port";
     private volatile boolean shutdown = false;
+    private static final ClassLoader BOOTSTRAP_CLASSLOADER;
+
+    static {
+        //Get list of files in bootstrap folder
+        final List<URL> urls = new ArrayList<>();
+        try {
+            Files.list(Paths.get("lib/bootstrap")).forEach(p -> {
+                try {
+                    urls.add(p.toUri().toURL());
+                } catch (final MalformedURLException mef) {
+                    logger.warn("Unable to load " + p.getFileName() + " due to " + mef, mef);
+                }
+            });
+        } catch (final IOException ex) {
+            logger.error("Unable to properly initialize bootstrap classloader due to " + ex, ex);
```

Comment:
Will fix that and will make the following changes which ensure RPM output aligns to tar output in terms of lib/ and lib/bootstrap contents


---
 nifi-assembly/pom.xml | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/nifi-assembly/pom.xml b/nifi-assembly/pom.xml
index d9f4f99..5eef82f 100755
--- a/nifi-assembly/pom.xml
+++ b/nifi-assembly/pom.xml
@@ -538,8 +538,8 @@ language governing permissions and limitations under the License. -->
                                                 <excludes>
                                                     <exclude>org.bouncycastle:bcprov-jdk15on</exclude>
                                                     <exclude>org.apache.commons:commons-lang3</exclude>
-                                                    <exclude>org.slf4j:jul-to-slf4j</exclude>
-                                                    <exclude>ch.qos.logback:logback-classic</exclude>
+                                                    <!-- must be in lib <exclude>org.slf4j:jul-to-slf4j</exclude> -->
+                                                    <!-- must be in lib <exclude>ch.qos.logback:logback-classic</exclude> -->
                                                     <exclude>javax.mail:mail</exclude>
                                                     <!-- must be in lib <exclude>org.apache.nifi:nifi-api</exclude> -->
                                                     <exclude>org.apache.nifi:nifi-bootstrap</exclude>
@@ -547,7 +547,7 @@ language governing permissions and limitations under the License. -->
                                                     <exclude>org.apache.nifi:nifi-processor-utils</exclude>
                                                     <!-- must be in lib <exclude>org.apache.nifi:nifi-properties</exclude> -->
                                                     <exclude>org.apache.nifi:nifi-properties-loader</exclude>
-                                                    <exclude>org.slf4j:slf4j-api</exclude>
+                                                    <!-- must be in lib <exclude>org.slf4j:slf4j-api</exclude> -->
                                                     <exclude>javax.activation:activation</exclude>
                                                     <exclude>org.antlr:antlr-runtime</exclude>
                                                     <exclude>asm:asm</exclude>
@@ -559,7 +559,7 @@ language governing permissions and limitations under the License. -->
                                                     <exclude>com.fasterxml.jackson.core:jackson-databind</exclude>
                                                     <exclude>com.jayway.jsonpath:json-path</exclude>
                                                     <exclude>net.minidev:json-smart</exclude>
-                                                    <exclude>ch.qos.logback:logback-core</exclude>
+                                                    <!-- must be in lib <exclude>ch.qos.logback:logback-core</exclude> -->
                                                     <exclude>org.apache.nifi:nifi-security-utils</exclude>
                                                     <exclude>org.apache.nifi:nifi-utils</exclude>
                                                     <!-- Items to not include which are also not in bootstrap -->
@@ -574,14 +574,14 @@ language governing permissions and limitations under the License. -->
                                                 <includes>
                                                     <include>org.bouncycastle:bcprov-jdk15on</include>
                                                     <include>org.apache.commons:commons-lang3</include>
-                                                    <include>org.slf4j:jul-to-slf4j</include>
+                                                    <!-- already in lib <include>org.slf4j:jul-to-slf4j</include> -->
                                                     <include>ch.qos.logback:logback-classic</include>
                                                     <include>javax.mail:mail</include>
                                                     <include>org.apache.nifi:nifi-api</include>
                                                     <include>org.apache.nifi:nifi-bootstrap</include>
                                                     <include>org.apache.nifi:nifi-expression-language</include>
                                                     <include>org.apache.nifi:nifi-processor-utils</include>
-                                                    <include>org.apache.nifi:nifi-properties</include>
+                                                    <!-- already in lib <include>org.apache.nifi:nifi-properties</include> -->
                                                     <include>org.apache.nifi:nifi-properties-loader</include>
                                                     <include>org.slf4j:slf4j-api</include>
                                                     <include>javax.activation:activation</include>
-- 
1.8.3.1